src/Features/Core/Portable/GoToDefinition/GoToDefinitionFeatureHelpers.cs

```diff
@@ -91,10 +91,10 @@ internal static class GoToDefinitionFeatureHelpers
 
             if (thirdPartyNavigationAllowed)
             {
-                var factory = solution.Services.GetService<IDefinitionsAndReferencesFactory>();
-                if (factory != null)
+                var provider = solution.Services.GetService<IExternalDefinitionItemProvider>();
+                if (provider != null)
                 {
-                    var thirdPartyItem = await factory.GetThirdPartyDefinitionItemAsync(solution, definitionItem, cancellationToken).ConfigureAwait(false);
+                    var thirdPartyItem = await provider.GetThirdPartyDefinitionItemAsync(solution, definitionItem, cancellationToken).ConfigureAwait(false);
```

Comment:
oh, i see. you just moved to having this be the single place where we try to call out.  that works too.